projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathDisplayHintsTest.java

```diff
@@ -0,0 +1,156 @@
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.Configuration.ConfigurationBuilder;
+import com.jayway.jsonpath.JsonPath;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.batfish.common.BatfishException;
+import org.batfish.common.util.BatfishObjectMapper;
+import org.batfish.common.util.CommonUtil;
+import org.batfish.datamodel.questions.DisplayHints;
+import org.batfish.datamodel.questions.DisplayHints.ExtractionHint;
+import org.batfish.question.jsonpath.JsonPathQuestionPlugin.JsonPathAnswerer;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class JsonPathDisplayHintsTest {
+
+  @Rule
+  public ExpectedException _thrown = ExpectedException.none();
+
+  DisplayHints _displayHints;
+
+  @Before
+  public void readDisplayHints() throws IOException {
+    String text =
+        CommonUtil.readResource("org/batfish/question/jsonpath/jsonPathDisplayHintsTestHints.json");
+    BatfishObjectMapper mapper = new BatfishObjectMapper();
+    _displayHints = mapper.readValue(text, DisplayHints.class);
+  }
+
+  private static JsonPathResult computeJsonPathResult(
+      String jsonFile, String path, boolean includeSuffix) {
+    Configuration.setDefaults(BatfishJsonPathDefaults.INSTANCE);
+    ConfigurationBuilder b = new ConfigurationBuilder();
+    Configuration baseConfiguration = b.build();
+    String jsonStr = CommonUtil.readResource(jsonFile);
+    Object jsonObject = JsonPath.parse(jsonStr, baseConfiguration).json();
+
+    JsonPathQuery query = new JsonPathQuery();
+    query.setPath(path);
+    query.setSuffix(includeSuffix);
+
+    JsonPathResult result = JsonPathAnswerer.computeResult(jsonObject, query);
+
+    return result;
+  }
+
+  private DisplayHints getDisplayHints(String displayVariable) {
+    DisplayHints displayHints = new DisplayHints();
+    Map<String,ExtractionHint> extractionHints = new HashMap<>();
+    extractionHints.put(displayVariable, _displayHints.getExtractionHints().get(displayVariable));
+    displayHints.setExtractionHints(extractionHints);
+    return displayHints;
+  }
+
+  @Test
+  public void jsonPathDisplayHintInterfaceListFromPrefixOfSuffixTest() throws IOException {
+    String displayVariable = "interfaceListFromPrefixOfSuffix";
+    JsonPathResult result =
+        computeJsonPathResult(
+            "org/batfish/question/jsonpath/jsonPathDisplayHintsTestObject.json",
+            "$.nodes[*].interfaces", true);
+
+    Map<String, Map<String, JsonNode>> displayValues =
+        result.computeDisplayValues(getDisplayHints(displayVariable));
+
+    String key1 = "'nodes'->'node1'->'interfaces'";
+    String key2 = "'nodes'->'node2'->'interfaces'";
+
+    assertThat(displayValues.size(), equalTo(2));
+    assertThat(displayValues.containsKey(key1), equalTo(true));
+    assertThat(displayValues.containsKey(key2), equalTo(true));
```

Comment:
shouldn't the key here be "interfaceListFromPrefixOfSuffix"? I thought it was taken from the name of the extraction hint, like `node1` in displayHintsTest.json

Reply:
you are likely misunderstanding the data structure layout. more tomorrow.